tf_yarn/client.py

```diff
@@ -122,7 +122,7 @@ def _setup_task_env(
         **env,
         # Make Python modules/packages passed via ``files`` importable.
         "PYTHONPATH": ".:" + env.get("PYTHONPATH", ""),
-        "PEX_ROOT": os.path.join("/tmp", str(uuid.uuid4()))
+        "PEX_ROOT": "./.pex"

```

Comment:
I'd prefer to not change the path for everyone as tf-yarn is open sourced.
What about:
task_env = {
        # Make Python modules/packages passed via ``files`` importable.
        "PYTHONPATH": ".:" + env.get("PYTHONPATH", ""),
        "PEX_ROOT": f"./{str(uuid.uuid4())}" if is_criteo() else os.path.join("/tmp", str(uuid.uuid4())),
        **env,
    }
I suggest:
- To redirect PEX_ROOT to the local dir by default only for Criteo
- To create a random directory as I'm not sure what would happen if several executors run one the same machine. Do you have the answer ?
- To let the user overwrite PEX_ROOT via the variable env (which is a parameter of the public API run_on_yarn)

What do you think ?

Reply:
Thanks for taking a look for the review !

> To redirect PEX_ROOT to the local dir by default only for Criteo

Good remark given the fact that is an open source project  !

> To create a random directory as I'm not sure what would happen if several executors run one the same machine. Do you have the answer ?

The reason why I didn't do this is because I have noticed PEX has already handled this by creating a folder looking like this:
`.pex/installed_wheels/28f26aef0732509f0407512380cb17a0f04451b0` , I didn't encounter a race condition problem by now. But I do agree that adding a random directory would be safer.

> To let the user overwrite PEX_ROOT via the variable env (which is a parameter of the public API run_on_yarn)
Indeed, I have hesitated on this as well, why not !

I will modify my PR according to these suggestions.

Reply:
@nateagr The commit has been updated accordingly, ready to be reviewed.